data/src/main/java/me/androidbox/data/di/NetworkModule.kt

```diff
@@ -0,0 +1,68 @@
+package me.androidbox.data.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.Reusable
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import me.androidbox.data.BuildConfig
+import me.androidbox.data.remote.interceptor.ApiKeyInterceptor
+import me.androidbox.data.remote.network.authentication.AuthenticationService
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
+
+    @Reusable
+    @Provides
+    fun provideHttpLoggingInteceptor(): HttpLoggingInterceptor {
+        val loggingInterceptor = HttpLoggingInterceptor()
+
+        loggingInterceptor.level = if (BuildConfig.DEBUG) {
+            HttpLoggingInterceptor.Level.BODY
+        } else {
+            HttpLoggingInterceptor.Level.NONE
+        }
+
+        return loggingInterceptor
+    }
+
+    @Reusable
+    @Provides
+    fun provideApiKeyInterceptor(): ApiKeyInterceptor {
+        return ApiKeyInterceptor()
+    }
+
+    @Reusable
+    @Provides
+    fun provideOkHttpClient(
+        loggingInterceptor: HttpLoggingInterceptor,
+        apiKeyInterceptor: ApiKeyInterceptor
+    ): OkHttpClient {
+        return OkHttpClient
+            .Builder()
+            .addInterceptor(loggingInterceptor)
+            .addInterceptor(apiKeyInterceptor)
+            .build()
+    }
+
+    @Reusable
+    @Provides
+    fun provideRetrofit(okHttpClient: OkHttpClient): Retrofit {
+        return Retrofit.Builder()
+            .baseUrl(BuildConfig.BUSBY_TASKY_API)
+            .client(okHttpClient)
+            .addConverterFactory(MoshiConverterFactory.create())
+            .build()
+    }
+
+    @Reusable
```

Comment:
Would it be best if these are `Reusable` as dagger will only create them if necessary. Or should they be `Singleton`?

Reply:
I personally use Singleton when im on a SingletonComponent